### PR TITLE
Add dozens of new integration tests to the GDScript test suite

### PR DESCRIPTION
**Pull request by Calinou**
This also makes all scripts use tab indentation (except the ones specifically designed to test space indentation or mixed tab/space indentation).

Reviewing tip: Use the file filter to hide `.out` files in the **Files changed** tab:

image

<details>
<summary>dictionary_consecutive_commas (test that currently crashes)</summary>

```gdscript
func test():
	# Dictionaries with consecutive commas are not allowed.
	var dictionary = { "hello": "world",, }

	dictionary = { hello = "world",, }
```

```
GDTEST_PARSER_ERROR
Expected expression as dictionary key.
```
</details>

### CODE REVIEW

**Comment by Xrayez** [comment]
Seems like static checks occasionally cause CI errors with `*.out` files, I think those files should be ignored from static checks eventually. I remedied this problem by stripping and appending the newline in the original PR, but seems like it's not enough to eliminate those errors completely.

**Comment by Calinou** [comment]
> Seems like static checks occasionally cause CI errors with `*.out` files, I think those files should be ignored from static checks eventually. I remedied this problem by stripping and appending the newline for each generated `.out` file in the original PR, but seems like it's not enough to eliminate those errors completely.

This should be fixed now.

**Comment by mhilbrunner** [comment]
Built and tested.

`tests/scripts/parser/errors/dictionary-consecutive-commas.gd` currently crashes the GDScript parser.

**Comment by Calinou** [comment]
> Built and tested.
> 
> `tests/scripts/parser/errors/dictionary-consecutive-commas.gd` currently crashes the GDScript parser.

Fixed :slightly_smiling_face:

**Comment by mhilbrunner** [comment]
@Calinou Whats the state of this? :)

**Comment by Calinou** [comment]
> @Calinou Whats the state of this? :)

It should be mergeable now. We can add more tests later as needed.

**Comment by mhilbrunner** [comment]
CI disagrees :P

**Comment by Calinou** [comment]
I've added many tests related to heavy nesting in `parser/features`.

File: modules/gdscript/tests/scripts/parser/errors/assignment_to_constant.out [hunk 1]
```diff
@@ -0,0 +1,2 @@
+GDTEST_PARSER_ERROR
+Expected variable name after "var".
```
Comment (vnen) [inline modules/gdscript/tests/scripts/parser/errors/assignment_to_constant.out:2]: This `.out` file don't have a corresponding `.gd` file.

File: modules/gdscript/tests/scripts/parser/features/constants.out [hunk 2]
```diff
@@ -0,0 +1,33 @@
+GDTEST_OK
+>> WARNING
+>> Line: 2
+>> UNUSED_LOCAL_CONSTANT
+>> The local constant '_TEST' is declared but never used in the block. If this is intended, prefix it with an underscore: '__TEST'
```
Comment (vnen) [inline modules/gdscript/tests/scripts/parser/features/constants.out:5]: This needs a fix in the GDScript code.

File: modules/gdscript/tests/scripts/parser/features/float_notation.out [hunk 3]
```diff
@@ -0,0 +1,73 @@
+GDTEST_OK
+>> WARNING
+>> Line: 3
+>> ASSERT_ALWAYS_TRUE
+>> Assert statement is redundant because the expression is always true.
```
Comment (vnen) [inline modules/gdscript/tests/scripts/parser/features/float_notation.out:5]: Given this warning isn't relevant for the test, I feel it's better to replace `assert` by `print`.

File: modules/gdscript/tests/scripts/parser/features/in.out [hunk 4]
```diff
@@ -0,0 +1,9 @@
+GDTEST_OK
+>> WARNING
+>> Line: 2
+>> ASSERT_ALWAYS_TRUE
+>> Assert statement is redundant because the expression is always true.
+>> WARNING
```
Comment (vnen) [inline modules/gdscript/tests/scripts/parser/features/in.out:6]: Same thing as above: `print` is probably better.

File: modules/gdscript/tests/scripts/parser/features/string_formatting.out [hunk 5]
```diff
@@ -0,0 +1,29 @@
+GDTEST_OK
+>> WARNING
+>> Line: 2
+>> ASSERT_ALWAYS_TRUE
+>> Assert statement is redundant because the expression is always true.
```
Comment (vnen) [inline modules/gdscript/tests/scripts/parser/features/string_formatting.out:5]: Same here.

File: modules/gdscript/tests/scripts/parser/features/constants.out (hunk 2 above)
Comment (Calinou) [reply modules/gdscript/tests/scripts/parser/features/constants.out:5]: I don't know how to fix that warning, should I ignore it or should I wait for a fix PR to be made first?

Comment (vnen) [reply modules/gdscript/tests/scripts/parser/features/constants.out:5]: I'd say leave like this for now, but open an issue about it.

**Review by vnen — APPROVED** [review]
LGTM. Thanks for working on this.

**Comment by akien-mga** [comment]
Thanks!

File: modules/gdscript/tests/scripts/analyzer/errors/bitwise_float_right_operand.gd [hunk 6]
```diff
@@ -0,0 +1,3 @@
+func test():
+	# Error here.
+	print(2 << 4.4)
```
Comment (kleonc) [inline modules/gdscript/tests/scripts/analyzer/errors/bitwise_float_right_operand.gd:3]: Should be `>>`.

Comment (Calinou) [reply modules/gdscript/tests/scripts/analyzer/errors/bitwise_float_right_operand.gd:3]: Good catch – I'll fix this in an upcoming PR :slightly_smiling_face:

**Pull request merged**